utilities/test_suite/HIP/Tensor_misc_hip.cpp

```diff
@@ -0,0 +1,643 @@
+                    return -1;
+                }
+                else if(qaMode && nDim == 3 && axisMask != 3 && (!computeMean || !computeStddev))
+                {
+                    std::cout<<"QA mode can only run with internal mean and stddev when nDim is 3"<<std::endl;
+                    return -1;
+                }
+
+                if (qaMode && nDim == 4 && (!computeMean && !computeStddev))
+                {
+                    std::cout<<"QA mode can only run with internal mean and stddev when nDim is 4"<<std::endl;
+                    return -1;
+                }
+
+                Rpp32u size = 1; // length of input tensors differ based on axisMask and nDim
+                Rpp32u maxSize = 1;
+                for(int batch = 0; batch < batchSize; batch++)
+                {
+                    size = 1;
+                    for(int i = 0; i < nDim; i++)
+                        size *= ((axisMask & (int)(pow(2,i))) >= 1) ? 1 : roiTensor[(nDim * 2 * batch) + nDim + i];
+                    maxSize = max(maxSize, size);
+                }
+                cout << "computed maxsize:" << maxSize << endl;
+
+                // allocate memory if not memory is allocated
+                if(meanTensor == nullptr)
+                    CHECK(hipMalloc(&meanTensor, maxSize * batchSize * sizeof(Rpp32f)));
+
+                if(stdDevTensor == nullptr)
+                    CHECK(hipMalloc(&stdDevTensor, maxSize * batchSize * sizeof(Rpp32f)));
+
+                if(!(computeMean && computeStddev))
+                {
+                    Rpp32f *meanTensorCPU = (Rpp32f *)malloc(maxSize * batchSize * sizeof(Rpp32f));
+                    Rpp32f *stdDevTensorCPU = (Rpp32f *)malloc(maxSize * batchSize * sizeof(Rpp32f));
+                    fill_mean_stddev_values(nDim, batchSize, size, meanTensorCPU, stdDevTensorCPU, qaMode);
+                    CHECK(hipMemcpy(meanTensor, meanTensorCPU, maxSize * batchSize * sizeof(Rpp32f), hipMemcpyHostToDevice));
+                    CHECK(hipMemcpy(stdDevTensor, stdDevTensorCPU, maxSize * batchSize * sizeof(Rpp32f), hipMemcpyHostToDevice));
+                    CHECK(hipDeviceSynchronize());
+                    free(meanTensorCPU);
+                    free(stdDevTensorCPU);
+                }
+
+                startWallTime = omp_get_wtime();
+                rppt_normalize_generic_gpu(d_inputF32, srcDescriptorPtrND, d_outputF32, dstDescriptorPtrND, axisMask, meanTensor, stdDevTensor, computeMean, computeStddev, scale, shift, roiTensor, handle);
+
+                // compare outputs if qaMode is true
+                if(qaMode)
+                {
+                    CHECK(hipDeviceSynchronize());
+                    CHECK(hipMemcpy(outputF32, d_outputF32, numValues * sizeof(Rpp32f), hipMemcpyDeviceToHost));
+                    CHECK(hipDeviceSynchronize());
+                    compare_output(outputF32, nDim, batchSize, dst, funcName, axisMask);
```

Comment:
Same here, cannot include all this in the time we measure. How have we been measuring time till now?

Reply:
This copy is done only in case of qaMode. In qaMode we won't display the times
`if(qaMode)
                {
                    CHECK(hipDeviceSynchronize());
                    CHECK(hipMemcpy(outputF32, d_outputF32, numValues * sizeof(Rpp32f), hipMemcpyDeviceToHost));
                    CHECK(hipDeviceSynchronize());
                    compare_output(outputF32, nDim, batchSize, dst, funcName, axisMask);
                }`